### PR TITLE
Create task from Talk message

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob Röhrl <jakob.roehrl@web.de>

**Todo**
- [ ] Style layout https://github.com/nextcloud/tasks/issues/1682
- [ ] Select the default task list instead of the first one https://github.com/nextcloud/tasks/issues/1687

---

**Screenshots:**
Screenshot 2021-06-24 at 11-02-54 Talk updates ✅ - Talk - Nextcloud

image

---
references:
https://github.com/nextcloud/deck/pull/2795

### CODE REVIEW

File: webpack.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
 const webpackConfig = require('@nextcloud/webpack-vue-config')
 
+const path = require('path')
+
+webpackConfig.entry = {
+	...webpackConfig.entry,
+	dashboard: path.join(__dirname, 'src', 'init-talk.js'),
```
Comment (raimund-schluessler) [inline webpack.js:7]:
```suggestion
	talk: path.join(__dirname, 'src', 'init-talk.js'),
```

Comment (jakobroehrl) [reply webpack.js:7]: I don't see the action after changing this, do you have another idea?

**Comment by raimund-schluessler** [comment]
@jakobroehrl The problem was that you never registered the `BeforeTemplateRenderedListener`, so you the Javascript file was never actually loaded. I fixed that, so now the file is loaded and an entry `Create new task` shows up in the Talk message action:

Screenshot 2021-06-24 at 11-02-54 Talk updates ✅ - Talk - Nextcloud

When you click it, the modal opens and throws a lot of errors. This is tbd yet.

**Comment by jakobroehrl** [comment]
Thanks a lot! :)
I work on the create dialog and maybe we can use it for the dashboard to add an task directly

**Comment by jakobroehrl** [comment]
@raimund-schluessler 
How to get all the calendars from the actual user into this file? Thanks!
https://github.com/nextcloud/tasks/blob/0ed0c6d98118622f32d2da1d9b5b99fcc9c11b9a/src/components/TaskCreateDialog.vue

Played a little bit around but couldn't figure it out.

**Comment by raimund-schluessler** [comment]
> 
> 
> @raimund-schluessler
> How to get all the calendars from the actual user into this file? Thanks!
> https://github.com/nextcloud/tasks/blob/0ed0c6d98118622f32d2da1d9b5b99fcc9c11b9a/src/components/TaskCreateDialog.vue

Similar to how you get them here:
https://github.com/nextcloud/tasks/blob/ed10468daf82b1cf4792f183da53b3ef62d4ddbf/src/views/AppSidebar.vue#L646-L647

But you also need to request them first, of course.
https://github.com/nextcloud/tasks/blob/ed10468daf82b1cf4792f183da53b3ef62d4ddbf/src/App.vue#L59-L60

**Comment by jakobroehrl** [comment]
@raimund-schluessler 
this.$store is undefined 
Tried to import it but it didn't work out.

**Comment by raimund-schluessler** [comment]
> 
> 
> @raimund-schluessler
> this.$store is undefined
> Tried to import it but it didn't work out.

Without seeing the whole code, it's difficult to know what's wrong. But I guess you didn't use `Vuex`. You can have a look at how it works for the dashboard:
https://github.com/nextcloud/tasks/blob/6a4dd9af562075881379062666ff7cb6e1c060f0/src/dashboard.js#L33-L35

**Comment by raimund-schluessler** [comment]
@jakobroehrl I fixed the error you were seeing after loading the calendars. The problem is that you cannot simply render an array of Objects with `{{ objectsArray }}` in your template. The calendars are now loaded and shown correctly.

Screenshot 2021-07-01 at 22-20-12 Talk updates ✅ - Talk - Nextcloud

There is still quite a bit to do:

- [ ] Creating a task is non-functional
- [ ] Task summary and description should be set from the talk message content
- [ ] Layout needs style fixes
- [ ] Code needs a cleanup
- [ ] Optional: Select the default task list instead of the first one

**Comment by raimund-schluessler** [comment]
>    * [ ]  Select the default task list instead of the first one: I always get the personal calendar here: https://github.com/nextcloud/tasks/blob/7bfe59e0dca63633b3f21a7ab6cef13099f8ac59/src/components/TaskCreateDialog.vue#L116-L120
>        Why?

The default calendar is not loaded here. Normally it comes from the initialState service, see https://github.com/nextcloud/tasks/blob/master/lib/Controller/PageController.php#L62
But here in Talk, this does not work, since the initial state is never loaded. So we need a separate endpoint in PHP to get only the default calendar and then query it in Javascript via axios.

**Comment by jakobroehrl** [comment]
> > * [ ]   Select the default task list instead of the first one: I always get the personal calendar here: https://github.com/nextcloud/tasks/blob/7bfe59e0dca63633b3f21a7ab6cef13099f8ac59/src/components/TaskCreateDialog.vue#L116-L120
> >   
> >   Why?
> 
> The default calendar is not loaded here. Normally it comes from the initialState service, see https://github.com/nextcloud/tasks/blob/master/lib/Controller/PageController.php#L62
> But here in Talk, this does not work, since the initial state is never loaded. So we need a separate endpoint in PHP to get only the default calendar and then query it in Javascript via axios.

OK, I have an idea how this could be solved. I would do it in another PR, OK?

Could you maybe look over this: 
I don't get an result here back: https://github.com/nextcloud/tasks/blob/7bfe59e0dca63633b3f21a7ab6cef13099f8ac59/src/components/TaskCreateDialog.vue#L167

**Comment by raimund-schluessler** [comment]
> Could you maybe look over this:
> I don't get an result here back:
> 
> https://github.com/nextcloud/tasks/blob/7bfe59e0dca63633b3f21a7ab6cef13099f8ac59/src/components/TaskCreateDialog.vue#L167

Looking at the `createTask` action in the store, I think it simply does not return anything. What do you need the Task for?

**Comment by jakobroehrl** [comment]
> > Could you maybe look over this:
> > I don't get an result here back:
> > https://github.com/nextcloud/tasks/blob/7bfe59e0dca63633b3f21a7ab6cef13099f8ac59/src/components/TaskCreateDialog.vue#L167
> 
> Looking at the `createTask` action in the store, I think it simply does not return anything. What do you need the Task for?

I want to give a feedback to the user (or close the create modal) if the task was created.

**Comment by raimund-schluessler** [comment]
> 
> 
> > > Could you maybe look over this:
> > > I don't get an result here back:
> > > https://github.com/nextcloud/tasks/blob/7bfe59e0dca63633b3f21a7ab6cef13099f8ac59/src/components/TaskCreateDialog.vue#L167
> > 
> > 
> > Looking at the `createTask` action in the store, I think it simply does not return anything. What do you need the Task for?
> 
> I want to give a feedback to the user (or close the create modal) if the task was created.

I guess you need to adjust the `createTask` action then.

**Comment by raimund-schluessler** [comment]
@jakobroehrl Please have a look. I think it works ok, but needs some style fixes.

**Comment by jakobroehrl** [comment]
> @jakobroehrl Please have a look. I think it works ok, but needs some style fixes.

It's working great! The looks a bit weired, but I didn't figure it out how to solve.
image

**Comment by raimund-schluessler** [comment]
I will merge this now as it is, since it works as it should. Style fixes will come with https://github.com/nextcloud/tasks/issues/1682 later.

**Pull request merged**